# Assignment 1

### Step 1. Download dataset and import necessary libraries.

In [1]:
import pandas as pd
import numpy as np

### Step 2. Assign variable to it.

In [2]:
telco = pd.read_csv('Telco-Customer-Churn.csv')
telco.head(3)

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ...  DeviceProtection  \
0  No phone service             DSL             No  ...                No   
1                No             DSL            Yes  ...               Yes   
2                No             DSL            Yes  ...                No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   

      PaymentMethod MonthlyCharges  TotalCharges Churn  
0  Electronic check          29.85         29.85    No  
1      Mailed check          56.95        1889.5    No  
2      Mailed check          53.85        108.15   Yes  

[3 rows x 21 columns]

### Step 3. Basic info about the DataFrame.

In [3]:
telco.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
customerID          7043 non-null object
gender              7043 non-null object
SeniorCitizen       7043 non-null int64
Partner             7043 non-null object
Dependents          7043 non-null object
tenure              7043 non-null int64
PhoneService        7043 non-null object
MultipleLines       7043 non-null object
InternetService     7043 non-null object
OnlineSecurity      7043 non-null object
OnlineBackup        7043 non-null object
DeviceProtection    7043 non-null object
TechSupport         7043 non-null object
StreamingTV         7043 non-null object
StreamingMovies     7043 non-null object
Contract            7043 non-null object
PaperlessBilling    7043 non-null object
PaymentMethod       7043 non-null object
MonthlyCharges      7043 non-null float64
TotalCharges        7043 non-null object
Churn               7043 non-null object
dtypes: float64(1), int64(2), obj

### Step 4. Basic statistics about the dataset's numeric columns.

In [4]:
telco.describe()

SeniorCitizen       tenure  MonthlyCharges
count    7043.000000  7043.000000     7043.000000
mean        0.162147    32.371149       64.761692
std         0.368612    24.559481       30.090047
min         0.000000     0.000000       18.250000
25%         0.000000     9.000000       35.500000
50%         0.000000    29.000000       70.350000
75%         0.000000    55.000000       89.850000
max         1.000000    72.000000      118.750000

### Step 5. Display first three observations and last three observations.

In [5]:
telco.iloc[0:3]

customerID  gender  SeniorCitizen Partner Dependents  tenure PhoneService  \
0  7590-VHVEG  Female              0     Yes         No       1           No   
1  5575-GNVDE    Male              0      No         No      34          Yes   
2  3668-QPYBK    Male              0      No         No       2          Yes   

      MultipleLines InternetService OnlineSecurity  ...  DeviceProtection  \
0  No phone service             DSL             No  ...                No   
1                No             DSL            Yes  ...               Yes   
2                No             DSL            Yes  ...                No   

  TechSupport StreamingTV StreamingMovies        Contract PaperlessBilling  \
0          No          No              No  Month-to-month              Yes   
1          No          No              No        One year               No   
2          No          No              No  Month-to-month              Yes   

      PaymentMethod MonthlyCharges  TotalCharges Churn  
0  Electronic check          29.85         29.85    No  
1      Mailed check          56.95        1889.5    No  
2      Mailed check          53.85        108.15   Yes  

[3 rows x 21 columns]

In [6]:
telco.loc[7040:7043]

customerID  gender  SeniorCitizen Partner Dependents  tenure  \
7040  4801-JZAZL  Female              0     Yes        Yes      11   
7041  8361-LTMKD    Male              1     Yes         No       4   
7042  3186-AJIEK    Male              0      No         No      66   

     PhoneService     MultipleLines InternetService OnlineSecurity  ...   \
7040           No  No phone service             DSL            Yes  ...    
7041          Yes               Yes     Fiber optic             No  ...    
7042          Yes                No     Fiber optic            Yes  ...    

     DeviceProtection TechSupport StreamingTV StreamingMovies        Contract  \
7040               No          No          No              No  Month-to-month   
7041               No          No          No              No  Month-to-month   
7042              Yes         Yes         Yes             Yes        Two year   

     PaperlessBilling              PaymentMethod MonthlyCharges  TotalCharges  \
7040              Yes           Electronic check          29.60        346.45   
7041              Yes               Mailed check          74.40         306.6   
7042              Yes  Bank transfer (automatic)         105.65        6844.5   

     Churn  
7040    No  
7041   Yes  
7042    No  

[3 rows x 21 columns]

### Step 6. Calculate the percentage of churn by Phone Service.

In [7]:
ps_churn = telco.groupby(['PhoneService','Churn']).Churn.count()
churn_per_ps = telco.groupby(['PhoneService']).PhoneService.count()
ps_churn/churn_per_ps*100

PhoneService  Churn
No            No       75.073314
              Yes      24.926686
Yes           No       73.290363
              Yes      26.709637
dtype: float64

### Step 7. Calculate the percentage of churn by Internet Service.

In [8]:
is_churn = telco.groupby(['InternetService','Churn']).Churn.count()
churn_per_is = telco.groupby(['InternetService']).InternetService.count()
is_churn/churn_per_is*100

InternetService  Churn
DSL              No       81.040892
                 Yes      18.959108
Fiber optic      No       58.107235
                 Yes      41.892765
No               No       92.595020
                 Yes       7.404980
dtype: float64

Compare to DSL, the churn rate of Fiber optic subscriber is higher which is 41.89%.

### Step 8. Discover mean monthly charges per Internet Service.

In [9]:
telco.groupby('InternetService').MonthlyCharges.mean()

InternetService
DSL            58.102169
Fiber optic    91.500129
No             21.079194
Name: MonthlyCharges, dtype: float64

The mean monthly charges of Fiber optic which is the highest indicates the high churn rate of Fiber optic.

### Step 9. Discover number of female customer who monthly charges more than 90.

In [10]:
a = telco[(telco.MonthlyCharges > 90) & (telco.gender == 'Female')]
len(a)

860

### Step 10. Calculate max and min of monthly charges by Internet Service.

In [11]:
telco.groupby('InternetService').MonthlyCharges.agg(['min', 'max'])

min     max
InternetService               
DSL              23.45   94.80
Fiber optic      67.75  118.75
No               18.25   26.90

### Step 11. Calculate churn of customer by Contract types.

In [12]:
telco.groupby(['Contract','Churn']).size()

Contract        Churn
Month-to-month  No       2220
                Yes      1655
One year        No       1307
                Yes       166
Two year        No       1647
                Yes        48
dtype: int64

Customers signed up Month-to-month Contract have high tendency to churn compare to customers signed up for one year and  two years Contract. 